File: ojuz/ioi/2014/Day2/gondola-templates/grader.c
#include <stdio.h>
#include <assert.h>
#include "gondola.h"

int gondolaSequence[100001];
int replacementSequence[250001];

int main()
{
  int i, n, tag;
  int nr; 
  assert(scanf("%d", &tag)==1);
  assert(scanf("%d", &n)==1);
  for(i=0;i< n;i++)
    assert( scanf("%d", &gondolaSequence[i]) ==1);
  
  switch (tag){
  case 1: case 2: case 3:
    printf("%d\n", valid(n, gondolaSequence));
    break;

  case 4: case 5: case 6:
    nr = replacement(n, gondolaSequence, replacementSequence);
    printf("%d ", nr);
    if (nr > 0)
      {
	for (i=0; i<nr-1; i++)
	  printf("%d ", replacementSequence[i]);
	printf("%d\n", replacementSequence[nr-1]);
      }  
    else printf("\n");
    break;

  case 7: case 8: case 9: case 10:
    printf("%d\n",  countReplacement(n, gondolaSequence));  
    break;
  }

  return 0;
}

File: ojuz/ioi/2014/Day2/gondola-templates/grader.pas
program main;
uses gondola;
const
   mmax1   = 100001;
   maxs132 = 250001;
var
   gondolaSequence    : array [0..mmax1] of longint;
   replacementSequence : array [0..maxs132] of longint;
   i, n, nr : longint; {integer is not big enough}
   T	    : integer;
begin
   (* Read Input *)
   read(T);
   read(n);
   for i:=1 to n do
   begin
      read(gondolaSequence[i-1]);
   end;
   case T of
     1,2,3    : (*subtask 1-3 *)
       writeln(valid(n, gondolaSequence));
     4,5,6    : (*subtask 4-6 *)
       begin
	  nr:=replacement(n, gondolaSequence, replacementSequence);
	  write(nr); write(' ');
	  if nr > 0 then 
	  begin
	     for i:= 0 to nr-2 do
	     begin
		write(replacementSequence[i]); write(' ');
	     end;
	     writeln(replacementSequence[nr-1]);
	  end   
	  else writeln();
       end;
     7,8,9,10 : (*subtask 7-10 *)
       begin
	  writeln(countReplacement(n, gondolaSequence));
       end;
   end; { case }
end.
